src/mappings/Competition/mapping.ts

```diff
@@ -64,17 +64,40 @@ export function handleNewSuggestion(event: NewSuggestion): void {
     competitionSuggestion.suggestionId = event.params._suggestionId;
     competitionSuggestion.proposal = event.params._proposalId.toHex();
     competitionSuggestion.descriptionHash = event.params._descriptionHash.toHex();
-    let ipfsData = ipfs.cat('/ipfs/' + event.params._descriptionHash.toHex());
-    if (ipfsData != null) {
-      competitionSuggestion.description = ipfsData.toString();
-    }
     competitionSuggestion.suggester = event.params._suggester;
     competitionSuggestion.totalVotes = BigInt.fromI32(0);
     competitionSuggestion.createdAt = event.block.timestamp;
+    getSuggestionIPFSData(competitionSuggestion);
+
     competitionSuggestion.save();
   }
 }
 
+export function getSuggestionIPFSData(suggestion: CompetitionSuggestion): CompetitionSuggestion {
```

Comment:
this is the same as getting proposal ipfs data.
why not use the same code base ?